R/as.SpatRaster.R

```diff
@@ -0,0 +1,57 @@
+#' as.SpatRaster
+#'
+#' Convert magclass object to a  object
+#'
+#'
+#' @param x MAgPIE object
+#' @param res spatial data resolution. If not provided it will be guessed.
+#' @return A  object
+#' @author Jan Philipp Dietrich
+#' @seealso \code{\link{getCoords}}
+#' @examples
+#'
+#' if (requireNamespace("raster", quietly = TRUE)) {
+#'    r <- raster::brick(ncols = 360, nrows = 180, nl = 4)
+#'    r[85:89, 176:179] <- (1:20 %*% t(1:4))
+#'    r[15:19, 76:79] <-   (10 + 1:20 %*% t(1:4))
+#'
+#'    names(r) <- c("y2000..bla", "y2001..bla", "y2000..blub", "y2001..blub")
+#'    m <- as.magpie(r)
+#'    r2 <- as.SpatRaster(m)
+#' }
+#' @export
+
+
+as.SpatRaster <- function(x, res = NULL) { # nolint
+  if (!is.magpie(x)) stop("Input is not a magpie object")
+  if (!requireNamespace("terra", quietly = TRUE)) stop("The package \"terra\" is required!")
+
+  .guessRes <- function(xy) {
+    .tmp <- function(x) {
+      return(suppressWarnings(min(diff(sort(unique(x))))))
```

Comment:
What warnings are suppressed here?

Reply:
uh, good question. This is actually a copy&paset from `as.RasterBrick` (copy&paste as the other function hopefully might get deprecated at some poin assuming a full switch to `terra`)

Reply:
Then I'd say this is a good time to check what happens without suppressWarnings, maybe it's no longer necessary? This will in any case not break anything that's already existing, as this is a new function, so there's no risk as far as I can tell

Reply:
ok, done